src/ol/renderer/canvas/VectorTileLayer.js

```diff
@@ -434,97 +434,97 @@ class CanvasVectorTileLayerRenderer extends CanvasTileLayerRenderer {
-          ) {
-            tile = this.renderedTiles[i];
-            if (tile.getState() === TileState.LOADED) {
-              const extent = tileGrid.getTileCoordExtent(tile.tileCoord);
-              if (
-                source.getWrapX() &&
-                projection.canWrapX() &&
-                !containsExtent(projectionExtent, extent)
-              ) {
-                wrapX(coordinate, projection);
-              }
-              break;
+          if (tile.getState() === TileState.LOADED) {
+            const extent = tileGrid.getTileCoordExtent(tile.tileCoord);
+            if (
+              source.getWrapX() &&
+              projection.canWrapX() &&
+              !containsExtent(projectionExtent, extent)
+            ) {
+              wrapX(coordinate, projection);
             }
-            tile = undefined;
+            break;
           }
+          tile = undefined;
         }
-        if (!tile || tile.loadingSourceTiles > 0) {
-          resolve([]);
-          return;
-        }
-        const extent = tileGrid.getTileCoordExtent(tile.wrappedTileCoord);
-        const corner = getTopLeft(extent);
-        const tilePixel = [
-          (coordinate[0] - corner[0]) / resolution,
-          (corner[1] - coordinate[1]) / resolution,
+      }
+      if (!tile || tile.loadingSourceTiles > 0) {
+        resolve([]);
+        return;
+      }
+      const extent = tileGrid.getTileCoordExtent(tile.wrappedTileCoord);
+      const corner = getTopLeft(extent);
+      const tilePixel = [
+        (coordinate[0] - corner[0]) / resolution,
+        (corner[1] - coordinate[1]) / resolution,
+      ];
+      /** @type {Array<import("../../Feature.js").FeatureLike>} */
+      const features = tile
+        .getSourceTiles()
+        .reduce(function (accumulator, sourceTile) {
+          return accumulator.concat(sourceTile.getFeatures());
+        }, []);
+      /** @type {ImageData|undefined} */
+      let hitDetectionImageData = tile.hitDetectionImageData[layerUid];
+      if (!hitDetectionImageData) {
+        const tileSize = toSize(
+          tileGrid.getTileSize(
+            tileGrid.getZForResolution(resolution, source.zDirection)
+          )
+        );
+        const rotation = this.renderedRotation_;
+        const transforms = [
+          this.getRenderTransform(
+            tileGrid.getTileCoordCenter(tile.wrappedTileCoord),
+            resolution,
+            0,
+            HIT_DETECT_RESOLUTION,
+            tileSize[0] * HIT_DETECT_RESOLUTION,
+            tileSize[1] * HIT_DETECT_RESOLUTION,
+            0
+          ),
         ];
-        const features = tile
-          .getSourceTiles()
-          .reduce(function (accumulator, sourceTile) {
-            return accumulator.concat(sourceTile.getFeatures());
-          }, []);
-        let hitDetectionImageData = tile.hitDetectionImageData[layerUid];
-        if (!hitDetectionImageData && !this.animatingOrInteracting_) {
```

Comment:
`this.animatingOrInteracting_` does not exist in `ol/renderer/canvas/VectorTileLayer`, so I removed it. Should be ok to ignore interactions as the hitdetect image is per tile. Not sure about animations.